src/Feature/Changelog/Feature_Changelog.rei

```diff
@@ -1,19 +1,36 @@
 open Oni_Core;
 open Revery.UI;
 
+type commit;
+
+type model;
+
+let initial: model;
+
 [@deriving show({with_path: false})]
 type msg =
   | PullRequestClicked(int)
-  | CommitHashClicked(string);
+  | CommitHashClicked(string)
+  | ChangeExpanded(commit)
+  | ChangeContracted(commit);
+
+type outmsg =
+  | Nothing
+  | URL(string);
+
+let update: (model, msg) => (model, outmsg);
 
 module View: {
   module Full: {
     let make:
       (
+        ~state: model,
         ~theme: ColorTheme.Colors.t,
         ~uiFont: UiFont.t,
         ~onPullRequestClicked: int => unit,
         ~onCommitHashClicked: string => unit,
+        ~onChangeExpanded: commit => unit,
+        ~onChangeContracted: commit => unit,
```

Comment:
You don't really need separate handlers for internal messages. Instead you can create a "scoped" dispatch function, like is done here:

https://github.com/onivim/oni2/blob/599e880bda42adeb9b69318c73ef30c77036de48/src/Store/Features.re#L237-L239

That also allows you to make the `msg` type abstract, leaking fewer internal details.

Reply:
I consolidated it in the latest commit. Let me know if that works!

Reply:
Yeah, that looks good! I'm wondering why you decided to cal it `dispatchMsg` instead of just `dispatch` as we use everywhere else though.

Reply:
You're right, I just didnt want to confuse it with a more general `dispatch` that takes `Action`s rather than the `msg`s. Sorry about that!

Reply:
Right. `Action` isn't available here though, so there shouldn't be any confusion. It's an intentional part of the architecture that these components should be self-contained and have a clear boundary so that you can easily compose them. And because each component should have roughly the same structure (model/msg/update/view) the architecture is in a sense fractal. And so you want it to have the same name because it suggests its role within this component. And also, if you're going to have separate names for every level of the fractal you're going to have to become very creative ;)

Reply:
> And also, if you're going to have separate names for every level of the fractal you're going to have to become very creative ;)

Haha very true! Thanks for the input. I changed it in the latest commit.